phabricatoremails/source.py

```diff
@@ -0,0 +1,80 @@
+import json
+import pathlib
+from dataclasses import dataclass
+from typing import Optional
+
+import requests
+from requests import RequestException
+
+
+class PhabricatorException(Exception):
+    """An error that occurred while communicating with Phabricator"""
+
+    pass
+
+
+@dataclass
+class PhabricatorSource:
+    """Fetches feed information directly from the Phabricator API"""
+
+    server: str
+    token: str
+    override_limit: Optional[int]
+
+    def _request(self, endpoint, **query_parameters):
+        query_parameters = {k: v for k, v in query_parameters.items() if v}
+        url = "{}/api/{}".format(self.server, endpoint)
+
+        body = {
+            "params": json.dumps(
+                {"__conduit__": {"token": self.token}, **query_parameters}
+            )
+        }
+
+        try:
+            result = requests.post(url, data=body)
+        except RequestException:
+            raise PhabricatorException("Could not communicate with Phabricator")
+
+        if result.status_code != 200:
+            raise PhabricatorException(
+                "Phabricator request had incorrect status of {}. "
+                "The full response is: {}".format(result.status_code, result.text,)
+            )
+
+        body = result.json()
+
+        if body["error_code"]:
+            raise PhabricatorException(
+                "Phabricator request had an error: {}".format(body)
+            )
+
+        return body
+
+    def fetch_feed_end(self):
+        body = self._request("feed.for_email.status")
+        return int(body["result"])
+
+    def fetch_next(self, since):
+        body = self._request(
+            "feed.for_email.query",
+            storyLimit=self.override_limit if self.override_limit else 100,
```

Comment:
There is a hardcoded value here, please move this to its own variable and give it a meaningful name.